modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -58,6 +58,518 @@ static inline DST add(SRC1 x, SRC2 y)
+    {
+        constexpr int nlanes = v_float32::nlanes;
+        for (; x < length; )
+        {
+            for (; x <= length - nlanes; x += nlanes)
+            {
+                v_float32 a = v_cvt_f32(vx_load_expand(reinterpret_cast<const short*>(&in1[x])));
+                v_float32 b = v_cvt_f32(vx_load_expand(reinterpret_cast<const short*>(&in2[x])));
+                vx_store(reinterpret_cast<float*>(&out[x]), a - b);
+            }
+
+            if (x < length && length >= nlanes)
+            {
+                x = length - nlanes;
+            }
+        }
+    }
+    else if (std::is_same<SRC, short>::value && std::is_same<DST, uchar>::value)
+    {
+        constexpr int nlanes = v_uint8::nlanes;
+        for (; x < length; )
+        {
+            for (; x <= length - nlanes; x += nlanes)
+            {
+                v_int16 a1 = vx_load(reinterpret_cast<const short*>(&in1[x]));
+                v_int16 a2 = vx_load(reinterpret_cast<const short*>(&in1[x + nlanes / 2]));
+                v_int16 b1 = vx_load(reinterpret_cast<const short*>(&in2[x]));
+                v_int16 b2 = vx_load(reinterpret_cast<const short*>(&in2[x + nlanes / 2]));
+
+                vx_store(reinterpret_cast<uchar*>(&out[x]), v_pack_u(a1 - b1, a2 - b2));
+            }
+
+            if (x < length && length >= nlanes)
+            {
+                x = length - nlanes;
+            }
+        }
+    }
+    else if (std::is_same<SRC, float>::value && std::is_same<DST, uchar>::value)
+    {
+        constexpr int nlanes = v_uint8::nlanes;
+        for (; x < length; )
+        {
+            for (; x <= length - nlanes; x += nlanes)
+            {
+                v_float32 a1 = vx_load(reinterpret_cast<const float*>(&in1[x]));
+                v_float32 a2 = vx_load(reinterpret_cast<const float*>(&in1[x + nlanes / 4]));
+                v_float32 a3 = vx_load(reinterpret_cast<const float*>(&in1[x + 2 * nlanes / 4]));
+                v_float32 a4 = vx_load(reinterpret_cast<const float*>(&in1[x + 3 * nlanes / 4]));
+
+                v_float32 b1 = vx_load(reinterpret_cast<const float*>(&in2[x]));
+                v_float32 b2 = vx_load(reinterpret_cast<const float*>(&in2[x + nlanes / 4]));
+                v_float32 b3 = vx_load(reinterpret_cast<const float*>(&in2[x + 2 * nlanes / 4]));
+                v_float32 b4 = vx_load(reinterpret_cast<const float*>(&in2[x + 3 * nlanes / 4]));
+
+                v_int16 res1 = v_pack(v_round(a1 - b1), v_round(a2 - b2)),
+                        res2 = v_pack(v_round(a3 - b3), v_round(a4 - b4));
+
+                vx_store(reinterpret_cast<uchar*>(&out[x]), v_pack_u(res1, res2));
+            }
+
+            if (x < length && length >= nlanes)
+            {
+                x = length - nlanes;
+            }
+        }
+    }
+}
+
+template<typename SRC, typename DST>
+static inline void addw_simd(const SRC in1[], const SRC in2[], DST out[], float _alpha, float _beta, float _gamma, int length, int& x)
```

Comment:
please change this function to have same structure as other ones. I.e. to make this a pure dispatcher calling implementation fuction with proper casted arguments

Reply:
this applies to other functions as well

Reply:
Outdated